src/types/widget.ts

```diff
@@ -129,11 +129,24 @@ export interface Widget {
    * during this widget's initialization and life time.
    */
   dispose?(options: DisposeOptions): SearchParameters | void;
-  getConfiguration?(previousConfiguration: SearchParameters): SearchParameters;
+  /**
+   * This function is required for a widget to be taken in account for routing.
+   * It will derive a uiState for this widget based on the existing uiState and
+   * the search parameters applied.
+   * @param uiState current state
+   * @param widgetStateOptions extra information to calculate uiState
+   */
   getWidgetState?(
     uiState: IndexUiState,
     widgetStateOptions: WidgetStateOptions
   ): IndexUiState;
+  /**
+   * This function is required for a widget to behave correctly when a URL is
```

Comment:
This is half true since we know use this function to sets the configuration too.

Reply:
Do you have a better description? “Also used to keep track (or merge) of refinements when other widgets are added/removed”?